### PR TITLE
Adds workaround for gzip compression regression

### LINKED ISSUE #2495: Regression: No gzip for search results in Armeria server
After moving to Armeria gzip was disabled with no way of enabling it back

* Gzip compression in Armeria has to be enabled explicitly, looks like it's not enabled yet
* JSON is highly compressible, sending 1MB responses over the network is better than 10MB
* Documentation points to `zipkin.yml`, which has documented compression options that are ignored. They seemingly configure Spring server, which is not covered when Armeria is used

Proposed fix:
```java
@Configuration public class SpringConfiguration {
    @Bean @Order(0) ArmeriaServerConfigurator httpResponseGzipCompressionConfigurator() {
        return serverBuilder -> serverBuilder.decorator(
                s -> new HttpEncodingService(s,
                        mt -> mt.belongsTo(MediaType.ANY_APPLICATION_TYPE) || mt.belongsTo(MediaType.ANY_TEXT_TYPE),
                        1024));
    }

}
```
Note: This does not allow one to configure it, we probably want to reuse "compression" parameters in `Zipkin.yml`

_Reproducible on 12.5.8 version_

### PR DESCRIPTION
**Pull request by codefromthecrypt**
It is costly to ship uncompressed json to browsers. This puts in a
stop-gap.

Thanks @timothybasanov for reporting

FYI @anuraaga @hyangtack we will remove this if addressed upstream

Fixes #2495

### CODE REVIEW

File: zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java [hunk 1]
```diff
@@ -133,36 +142,44 @@ String processedIndexHtml(Resource indexHtml) {
...
-      .service("/zipkin/index.html", indexSwitchingService)
+        .asService());
+
+      sb.serviceUnder("/zipkin/", uiFileService);
 
       // TODO This approach requires maintenance when new UI routes are added. Change to the following:
       // If the path is a a file w/an extension, treat normally.
       // Otherwise instead of returning 404, forward to the index.
       // See https://github.com/twitter/finatra/blob/458c6b639c3afb4e29873d123125eeeb2b02e2cd/http/src/main/scala/com/twitter/finatra/http/response/ResponseBuilder.scala#L321
-      .service("/zipkin/", indexSwitchingService)
```
Comment (codefromthecrypt) [inline zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java:157]: don't worry about the red.. I was just re-organizing as we no longer have to fluent everything

File: zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/CompressionProperties.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * the License.
+ */
+package zipkin2.autoconfigure.ui;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.web.server.Compression;
+
+@ConfigurationProperties("server")
+class CompressionProperties {
+  public Compression getCompression() {
```
Comment (anuraaga) [inline zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/CompressionProperties.java:21]: Not a big deal but I presume spring already exposes a bean with this

Comment (codefromthecrypt) [reply zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/CompressionProperties.java:21]: surprisingly no.. I tried :)

**Comment by codefromthecrypt** [comment]
verified the browser moves less bytes now..

Screen Shot 2019-04-12 at 5 34 33 PM

**Comment by codefromthecrypt** [comment]
cutting this shortly as I need to release a guava fix anyway

**Pull request merged**